x-pack/docs/en/security/authentication/jwt-realm.asciidoc

```diff
@@ -0,0 +1,374 @@
+* `metadata` claims: JWTs claims in the form of strings, numbers, booleans, and collections are used as the {es} user's metadata, formatted as `metadata.jwt_claim_<key>` = `<value>` pairs.
+
+See <<ref-jwt-settings>>.
+
+===== Example JWT Realm Role Mapping Rule
+
+Example JWT realm role mapping rule.
+
+[source,console]
+----
+PUT /_security/role_mapping/jwt1_users
+{
+  "roles" : [ "user" ],
+  "rules" : { "all" : [
+      { "field": { "realm.name": "jwt1" } },
+      { "field": { "username": "principalname1" } }
+      { "field": { "dn": "CN=Principal Name 1,DC=example.com" } }
+      { "field": { "groups": "group1" } }
+      { "field": { "metadata.jwt_claim_other": "other1" } }
+  ] },
+  "enabled": true
+}
+----
+
+===== JWT realm authorization via authorization delegation
+
+.If Delegated Realms are used for authorization, this claim is available for roles lookup.
+* `principal` claim: JWT claim in the form of a string is used as the {es} user's principal value.
+
+See <<ref-jwt-settings>>.
+
+WARNING: When Delegated Authorization is enabled in JWT realm, claims for `groups`, `dn`, and `metadata` not used for the {es} user's values.
+         Only the JWT `principal` claim is passed through to Delegated Authorization realms.
+         The Delegated Authorization realms become responsible for populating all of the {es} user's values, not the JWT realm.
+
+====== Example JWT realm delegated authorization:
+
+Example JWT realm using Delegated Authorization to multiple other realms.
+
+[source,yaml]
+----
+xpack.security.authc.realms.jwt.jwt2.authorization_realms: file1,native1,ldap1,ad1
+----
+
+Example role mapping in realm `native1` for principal `principalname1`.
+
+[source,console]
+----
+PUT /_security/role_mapping/native1_users
+{
+  "roles" : [ "user" ],
+  "rules" : { "all" : [
+      { "field": { "realm.name": "native1" } },
+      { "field": { "username": "principalname1" } }
+  ] },
+  "enabled": true
+}
+----
+
+If realm `jwt2` successfully authenticates a client, and a JWT for principal `principalname1`, and Delegates Authorization to realm `native1`,
+then realm `native1` can lookup that {es} user's values, including using this role mapping rule linked to realm `native1`.
+
+[[hmac-oidc-example]]
+==== JWT Realm Example with OIDC HMAC Key
+
+These settings are for a JWT issuer, Elasticsearch, and a client of Elasticsearch.
+The example HMAC key is in OIDC HMAC compatible format; key bytes are UTF-8 encoded UNICODE characters.
+
+WARNING: HMAC UTF-8 keys need to be longer than HMAC random byte keys to achieve same key strength.
+
+[[hmac-oidc-example-jwt-issuer]]
+===== JWT issuer
+
+These values are for a bespoke authentication service which issues JWTs; external to {es}.
+
+[source,text]
+----
+Issuer:     iss8
+Audiences:  aud8
+Algorithms: HS256
+HMAC OIDC:  hmac-oidc-key-string-for-hs256-algorithm
+----
+
+[[hmac-oidc-example-jwt-realm]]
+===== JWT realm clear settings
```

Comment:
```suggestion
===== JWT realm static settings
```
Should we change `clear` to `static` here for consistency?

Fabio clarified yesterday that the [realm settings](https://elasticsearch_85189.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/security-settings.html#ref-jwt-settings) reference page calls settings `static` vs `secure`.

Reply:
What does "clear" means here? Could the header just be "JWT realm settings" instead?

Reply:
I think that **JWT realm settings** works here.